src/Aspire.Dashboard/Model/ResourceSourceViewModel.cs

```diff
@@ -18,6 +18,12 @@ public class ResourceSourceViewModel(string value, string? contentAfterValue, st
 
         if (resource.TryGetExecutableArguments(out var arguments))
         {
+            if (resource.TryGetProjectArguments(out var projectArgs))
+            {
+                var foundHostArgCount = arguments.TakeWhile((arg, i) => i < projectArgs.Length && string.Equals(arg, projectArgs[i], StringComparison.Ordinal)).Count();
```

Comment:
If I understand this correctly, the annotation in `DcpExecutor` has new code to add the existing args to `project.args`, only for them to be carefully removed here. Why not simply add only the args you want to show to the annotation?

At the least, please document what's going on here, as it seems like the logic in the dashboard is coupled to some behaviour in the app host, and it's not obvious why that is.

Reply:
+1 on @drewnoakes suggestion to "invert" the logic and put into the annotation only the arguments that should show up in the dashboard.

Other than this, I see no issues with this change

Reply:
Thank you both! I've updated this to put an annotation on the dcp resource of the launch arguments. I also included information on whether they contained secrets

Reply:
LGTM on that front